### PR TITLE
LPD-91791 Honor configured columns when exporting audit events

### PR DESCRIPTION
**Pull request by caio-farias**
https://liferay.atlassian.net/browse/LPD-91791

## Why

The audit event CSV export emitted a malformed row format and ignored the columns configured in the CSV Logging Message Formatter, so exports did not reflect the administrator's configuration.

## Key changes

- Drive export columns from `CSVLogMessageFormatterConfiguration`, keyed by its column names, so the CSV matches the configured columns and order.
- Reformat each row with `CSVUtil.encode` joined by commas, replacing the previous malformed quoting.
- Encode null field values as empty cells via `GetterUtil.getString`, preventing shifted columns.
- Extract `_buildCSV` and add unit tests covering the format and null handling.

### CODE REVIEW

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java [hunk 1]
```diff
@@ -81,6 +105,55 @@ protected void doServeResource(
...
+								_functions.get(column);
+
+							return CSVUtil.encode(
+								GetterUtil.getString(
+									function.apply(auditEvent)));
+						},
+						String.class),
+					StringPool.COMMA));
+
+			sb.append(StringPool.NEW_LINE);
```
[the remark this answers is not in the document]
Comment (ChrisKian) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:144]: Hey @caio-farias, this was one of the issues Brian was talking about. I'm not sure how it's round about, but maybe this is something he's looking for:

		for (AuditEvent auditEvent : auditEvents) {
			for (String column : columns) {

				Function<AuditEvent, String> function =
					_functions.get(column);

				sb.append(CSVUtil.encode(
					GetterUtil.getString(
						function.apply(auditEvent))));

				sb.append(StringPool.COMMA);
			}
			sb.setIndex(sb.index() - 1);

			sb.append(StringPool.NEW_LINE);

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java [hunk 2]
```diff
@@ -132,84 +206,80 @@ private String _getAuditEventsCSV(
...
-						_functions.values(),
-						function -> function.apply(auditEvent)),
-					"\", \""));
+		if (user == null) {
+			return StringPool.BLANK;
+		}
 
-			sb.append(StringPool.NEW_LINE);
+		return user.getEmailAddress();
+	}
```
[the remark this answers is not in the document]
Comment (ChrisKian) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:224]: This could be useful

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java [hunk 3]
```diff
@@ -132,84 +206,80 @@ private String _getAuditEventsCSV(
...
-			progressTracker.setPercent(percentage);
+		if (user == null) {
+			return StringPool.BLANK;
 		}
 
-		progressTracker.finish(resourceRequest);
-
-		return sb.toString();
+		return user.getScreenName();
 	}
```
[the remark this answers is not in the document]
Comment (ChrisKian) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:234]: Same here

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java [hunk 4]
```diff
@@ -132,84 +206,80 @@ private String _getAuditEventsCSV(
...
 	private static final Log _log = LogFactoryUtil.getLog(
 		ExportAuditEventsMVCResourceCommand.class);
 
+	private volatile String[] _columns;
 	private final LinkedHashMap<String, Function<AuditEvent, String>>
 		_functions =
 			LinkedHashMapBuilder.<String, Function<AuditEvent, String>>put(
-				"event-id",
-				auditEvent -> String.valueOf(auditEvent.getAuditEventId())
+				"additionalInfo", AuditEvent::getAdditionalInfo
```
Comment (ChrisKian) [inline modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:243]: We're changing the key format from `camelCase` (TextFormatter "E") to `hyphen-between` (TextFormatter "J"), is this intentional?

**Review by ChrisKian — CHANGES_REQUESTED** [review]

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java (hunk 4 above)
Comment (caio-farias) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:243]: Yes. Since the core idea of this ticket is to use `CSVLogMessageFormatterConfiguration` as the source configuration for building the CSV, using its pattern (camelCase one) on the column names of the CSV seems to be more consistent.

The TextFormatter would also work. But it would still need to address some cases like: 
- className -> resource-name
- classPK -> resource-id
- timestamp -> create-date
- eventType -> resource-action

So the map would still be necessary to address these cases.

File: modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (caio-farias) [reply modules/dxp/apps/portal-security-audit/portal-security-audit-web/src/main/java/com/liferay/portal/security/audit/web/internal/portlet/action/ExportAuditEventsMVCResourceCommand.java:224]: I've checked on that and this seems to be correct. At startup, the amount of audit events that are generated by guest and unauthenticated users can are reasonable, so I believe it might be a very common case in general.
